### PR TITLE
support obsolete postgres json format

### PR DESCRIPTION
**Pull request by seamusabshere**
because, in fact, people still use it :D

### CODE REVIEW

File: dbcrossbarlib/src/schema.rs [hunk 1]
```diff
@@ -126,9 +126,10 @@ pub enum DataType {
     Int32,
     /// 8-byte integer.
     Int64,
-    /// JSON data. This includes both Postgres `json` and `jsonb` types, the
-    /// differences between which don't usually matter when converting schemas.
+    /// JSON data. This includes Postgre's modern JSONB type, which is always preferred.
     Json,
+    /// JSON in Postgres's old format.
+    OldPgJson,
```
Comment (emk) [inline dbcrossbarlib/src/schema.rs]: Here, you've taken the portable, cross-database `DataType` enum, and added a new `DataType::OldPgJson` that only applies to a single database. Honestly, this feels really dubious: The portable, standardized data type should never include weird database-specific types, and certainly not obsolete ones.

So I think we need to go back to the drawing board, and design a testcase that shows what you want to accomplish. Then from there, we can figure out a clean way to do it, or at least hide any of the necessary hacks in the `postgres` driver itself, and not in the interchange types. Or maybe we should bring back `DataType::Custom(String)`, though I'm less certain of that.

In general, if you _really_ need to produce a PostgreSQL table containing weird things like legacy `json`, `oid`s, or other highly-specialized types, it's a good time to step back to the drawing board.

**Comment by emk** [comment]
I think in general that there are two things we'd ideally like to support:

1. Extract data from legacy `json` columns as regular portable `DataType::Json`.
2. Insert `DataType::Json` data into _existing_ `json` columns in already-created tables.

I'm not convinced that it's worthwhile to support creating _new_ tables with legacy `json` columns. The universal simplifying assumption of `dbcrossbar` is that we focus on best practices and "reasonable" data types, and we don't try to support the full set of data types of a specific database.

**Comment by emk** [comment]
Besides (1) and (2) above, can you think of any other use cases for legacy `json` types?

**Comment by seamusabshere** [comment]
@emk agreed in general, the only use case i need to support here is copying INTO existing json columns.

**Comment by emk** [comment]
This branch contains a variety of different fixes, including at least:

1. Fixes for strange column names. (Fixed in #66.)
2. A workaround for legacy `json` columns. (Fixed differently in #67.)
3. Disabling of BigQuery UPSERT checks. The recommended workaround for this is to edit the schema to mark all merge key fields as NOT NULL, which will inform `dbcrossbar` that you're taking responsibility for the issue.

Since the major features in this branch are fixed elsewhere (or have workarounds) and there's no plan to make this branch mergeable, I'm going to close this PR. Please feel free to re-open it if you want.

**Pull request closed without merging**